project_precommit_check

```diff
@@ -177,7 +177,13 @@ def is_correct_swift_version(swiftc, compatibility_version):
 
 
 def check(project, swift_branch, swiftc, compatibility_version):
-    if platform.system() == 'Darwin':
+    # Please, rewrite this script.
+    # Writing into function parameters is a bad manner. You can do better.
+    # Please, rewrite it.
+    # Thanks!
+
+    # find pre-installed if we don't specify parameter --swiftc
+    if platform.system() == 'Darwin' and not swiftc:
```

Comment:
This is not sufficient to support `--swiftc` flag on Darwin. The check in `is_correct_swift_version` will fail when trying to use a swiftc that doesn't match the compatibility configuration.